pkg/virt-handler/isolation/isolation.go

```diff
@@ -164,43 +168,81 @@ func (s *socketBasedIsolationDetector) AdjustResources(vm *v1.VirtualMachineInst
 		return nil
 	}
 
-	// bump memlock ulimit for libvirtd
-	res, err := s.Detect(vm)
+	isolationResult, err := s.Detect(vm)
+	if err != nil {
+		return err
+	}
+
+	// make the best estimate for memory required by libvirt
+	memlockSize, err := getMemlockSize(vm)
 	if err != nil {
 		return err
 	}
-	launcherPid := res.Pid()
 
 	processes, err := ps.Processes()
 	if err != nil {
 		return fmt.Errorf("failed to get all processes: %v", err)
 	}
 
-	for _, process := range processes {
-		// consider all processes that are virt-launcher children
-		if process.PPid() != launcherPid {
-			continue
+	// bump libvirtd and qemu processes locked memory limits
+	libvirtProcesses := filterVirtLaunchersLibvirtProcesses(isolationResult, processes)
+	for _, process := range libvirtProcesses {
+		err = setProcessMemoryLockLimit(process.Pid(), uint64(memlockSize), uint64(memlockSize))
+		if err != nil {
+			return fmt.Errorf("failed to set rlimit for memory lock: %v", err)
 		}
+	}
 
-		// libvirtd process sets the memory lock limit before fork/exec-ing into qemu
-		if process.Executable() != "libvirtd" {
-			continue
-		}
+	return nil
+}
 
-		// make the best estimate for memory required by libvirt
-		memlockSize, err := getMemlockSize(vm)
-		if err != nil {
-			return err
+func filterVirtLaunchersLibvirtProcesses(isolationResult IsolationResult, processes []ps.Process) []ps.Process {
+	var filteredProcesses []ps.Process
```

Comment:
Why do you need this? On line 208 you can just use `:=`, no?

Reply:
It may look redundant here because there are only two assignments.
I think its easier to track where a variable was declared when there are more then one assignments (the first is always `:=`).